### PR TITLE
Rover: pivot turn wobble fix

### PR DESCRIPTION
**Pull request by rmackay9**
This PR adds feed-foward support to the steering rate controller which resolves the wobble seen in this video: https://www.youtube.com/watch?v=dbgXc3vo5qw

This PR also resolves a few other minor issues:

- add feed-forward support to the speed-to-throttle controller although it is unlikely this will ever be used.
- fix turn-rate controller timeout and increase the timeout to 200ms
- allow the steering rate and speed-to-throttle controller's input filter to be set to zero (i.e. disabled)

### CODE REVIEW

**Comment by rmackay9** [comment]
This has been successfully tested by Kelly Shrock..

File: libraries/APM_Control/AR_AttitudeControl.cpp [hunk 1]
```diff
@@ -207,7 +207,7 @@ float AR_AttitudeControl::get_steering_out_rate(float desired_rate, bool skid_st
     // calculate dt
     const uint32_t now = AP_HAL::millis();
     float dt = (now - _steer_turn_last_ms) / 1000.0f;
-    if ((_steer_turn_last_ms == 0) || (dt > AR_ATTCONTROL_TIMEOUT_MS)) {
+    if ((_steer_turn_last_ms == 0) || (dt > (AR_ATTCONTROL_TIMEOUT_MS / 1000.0f))) {
```
Comment (peterbarker) [inline libraries/APM_Control/AR_AttitudeControl.cpp]: Maybe change the define to be in seconds instead?

File: libraries/AC_PID/AC_PID.cpp [hunk 2]
```diff
@@ -201,14 +202,15 @@ void AC_PID::save_gains()
 }
 
 /// Overload the function call operator to permit easy initialisation
-void AC_PID::operator() (float p, float i, float d, float imaxval, float input_filt_hz, float dt)
+void AC_PID::operator() (float p, float i, float d, float imaxval, float input_filt_hz, float dt, float ffval)
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AC_PID/AC_PID.cpp]: I originally named it "ff" but the compiler complained that it showed the "ff" parameter.

File: libraries/APM_Control/AR_AttitudeControl.h [hunk 3]
```diff
@@ -7,18 +7,19 @@
 
 // attitude control default definition
 #define AR_ATTCONTROL_STEER_ANG_P       1.00f
+#define AR_ATTCONTROL_STEER_RATE_FF     0.20f
 #define AR_ATTCONTROL_STEER_RATE_P      1.00f
 #define AR_ATTCONTROL_STEER_RATE_I      0.50f
 #define AR_ATTCONTROL_STEER_RATE_IMAX   1.00f
 #define AR_ATTCONTROL_STEER_RATE_D      0.00f
-#define AR_ATTCONTROL_STEER_RATE_FILT   5.00f
+#define AR_ATTCONTROL_STEER_RATE_FILT   50.00f
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/APM_Control/AR_AttitudeControl.h]: yes, it pretty much removes the filtering on inputs to the steering rate controller.  I don't think we know yet what are good values but in general, higher values (i.e. less filtering) are more likely to produce good control at the expensive of some jitter.

**Comment by rmackay9** [comment]
Thanks for the review.  I've modified get_throttle_out_speed to use the same timeout as the rest of the calls.

File: libraries/APM_Control/AR_AttitudeControl.cpp (hunk 1 above)
Comment (rmackay9) [reply libraries/APM_Control/AR_AttitudeControl.cpp]: I guess the timeout check is done in two slightly different ways - one method uses seconds, the other milliseconds... so consolidating the method might be a good thing to do..

**Pull request merged**